File: wrappers/python/tests/utils/wallet.py
from indy import wallet

async def create_and_open_wallet(pool_name="pool_1", wallet_name="wallet_1"):
    await wallet.create_wallet(pool_name, wallet_name, None, None, None)
    wallet_handle = await wallet.open_wallet(wallet_name, None, None)
    assert wallet_handle is not None
    return wallet_handle


async def close_wallet(wallet_handle):
    await wallet.close_wallet(wallet_handle)

File: libnullpay/tests/utils/wallet.rs
use indy::IndyError;
use indy::wallet;
use indy::WalletHandle;
use indy::future::Future;

pub const DEFAULT_WALLET_CREDENTIALS: &'static str = r#"{"key":"8dvfYSt5d1taSd6yJdpjq4emkwsPDDLYxkNFysFD2cZY", "key_derivation_method":"RAW"}"#;

pub fn create_wallet(config: &str) -> Result<(), IndyError> {
    wallet::create_wallet(config, DEFAULT_WALLET_CREDENTIALS).wait()
}

pub fn open_wallet(config: &str) -> Result<WalletHandle, IndyError> {
    wallet::open_wallet(config, DEFAULT_WALLET_CREDENTIALS).wait()
}

pub fn create_and_open_wallet() -> Result<WalletHandle, IndyError> {
    let wallet_name = format!("default-wallet-name-{}", super::sequence::get_next_id());
    let config = format!(r#"{{"id":"{}"}}"#, wallet_name);

    create_wallet(&config)?;
    open_wallet(&config)
}

pub fn close_wallet(wallet_handle: WalletHandle) -> Result<(), IndyError> {
    wallet::close_wallet(wallet_handle).wait()
}